src/Internal/HttpClientCall.php

```diff
@@ -0,0 +1,82 @@
+<?php
+declare(strict_types=1);
+
+namespace Retrofit\Internal;
+
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Retrofit\Call;
+use Retrofit\Callback;
+use Retrofit\Converter\ResponseBodyConverter;
+use Retrofit\HttpClient;
+use Retrofit\Response;
+use RuntimeException;
+use Throwable;
+
+readonly class HttpClientCall implements Call
+{
+    public function __construct(
+        private HttpClient $httpClient,
+        private RequestInterface $request,
+        private ?ResponseBodyConverter $responseBodyConverter,
+        private ?ResponseBodyConverter $errorBodyConverter
+    )
+    {
+    }
+
+    public function execute(): Response
+    {
+        $response = $this->httpClient->send($this->request());
+        return $this->createResponse($response);
+    }
+
+    public function enqueue(Callback $callback): Call
+    {
+        $this->httpClient->sendAsync(
+            $this->request(),
+            fn(ResponseInterface $response) => $callback->onResponse($this, $this->createResponse($response)),
+            fn(Throwable $throwable) => $callback->onFailure($this, $throwable)
+        );
+        return $this;
+    }
+
+    public function wait(): void
+    {
+        $this->httpClient->wait();
+    }
+
+    public function request(): RequestInterface
+    {
+        return $this->request;
+    }
+
+    private function createResponse(ResponseInterface $response): Response
+    {
+        $code = $response->getStatusCode();
+        $body = $response->getBody();
+
+        if ($code >= 200 && $code < 300) {
```

Comment:
Magic numbers. I'd prefer to extract this `$code >= 200 && $code < 300` to some other method with meaningful name.